src/flux/shape.py

```diff
@@ -198,9 +198,6 @@ def get_direct_irradiance(self, F0, Dsun, basemesh=None, eps=None):
             directions.
 
         '''
-        if basemesh == None:
```

Comment:
I don't think that removing this is correct. We still need to set the basemesh to self. We can squeeze it inside the second "if" though.